tests/test_write_to_catalog.py

```diff
@@ -0,0 +1,139 @@
+from tests.file_utils import json_dict_to_file_object
+from tests.stac_objects import MINIMAL_VALID_STAC_CATALOG_OBJECT
+
+
+@mark.infrastructure
+def should_create_new_root_catalog_if_doesnt_exist(subtests: SubTests, s3_client: S3Client) -> None:
+
+    with Dataset() as dataset, S3Object(
+        file_object=json_dict_to_file_object(
+            {
+                **deepcopy(MINIMAL_VALID_STAC_CATALOG_OBJECT),
+                STAC_ID_KEY: dataset.dataset_id,
+                STAC_TITLE_KEY: dataset.title,
+            }
+        ),
+        bucket_name=ResourceName.STORAGE_BUCKET_NAME.value,
+        key=f"{dataset.dataset_prefix}/{CATALOG_KEY}",
+    ):
+
+        body = {"Records": [{"body": dataset.dataset_prefix}]}
+
+        try:
+            task.lambda_handler(body, any_lambda_context())
+
+            expected_links: JsonList = [
+                {
+                    STAC_REL_KEY: STAC_ROOT_KEY,
+                    STAC_HREF_KEY: f"./{CATALOG_KEY}",
+                    STAC_TYPE_KEY: "application/json",
+                },
+                {
+                    STAC_REL_KEY: STAC_CHILD_KEY,
+                    STAC_HREF_KEY: f"./{dataset.dataset_prefix}/{CATALOG_KEY}",
+                    STAC_TYPE_KEY: "application/json",
+                },
+            ]
+            with smart_open(
+                f"s3://{ResourceName.STORAGE_BUCKET_NAME.value}/{CATALOG_KEY}"
+            ) as new_root_metadata_file:
+                catalog_json = load(new_root_metadata_file)
+
+                with subtests.test(msg="catalog title"):
+                    assert catalog_json[STAC_TITLE_KEY] == ROOT_CATALOG_TITLE
+
+                with subtests.test(msg="catalog description"):
+                    assert catalog_json[STAC_DESCRIPTION_KEY] == ROOT_CATALOG_DESCRIPTION
+
+                with subtests.test(msg="catalog links"):
+                    assert catalog_json[STAC_LINKS_KEY] == expected_links
+
+        finally:
+            delete_s3_key(ResourceName.STORAGE_BUCKET_NAME.value, CATALOG_KEY, s3_client)
+
+
+@mark.infrastructure
+def should_update_existing_root_catalog(subtests: SubTests) -> None:
+
+    with Dataset() as dataset, S3Object(
+        file_object=json_dict_to_file_object(
+            {
+                **deepcopy(MINIMAL_VALID_STAC_CATALOG_OBJECT),
+                STAC_ID_KEY: dataset.dataset_id,
+                STAC_TITLE_KEY: dataset.title,
+            }
+        ),
+        bucket_name=ResourceName.STORAGE_BUCKET_NAME.value,
+        key=f"{dataset.dataset_prefix}/{CATALOG_KEY}",
+    ), S3Object(
+        file_object=json_dict_to_file_object(
+            {
+                **deepcopy(MINIMAL_VALID_STAC_CATALOG_OBJECT),
+                STAC_ID_KEY: ROOT_CATALOG_ID,
+                STAC_DESCRIPTION_KEY: ROOT_CATALOG_DESCRIPTION,
+                STAC_TITLE_KEY: ROOT_CATALOG_TITLE,
+            }
+        ),
+        bucket_name=ResourceName.STORAGE_BUCKET_NAME.value,
+        key=CATALOG_KEY,
+    ):
```

Comment:
I think this test would be stronger if the original root catalogue already included some links, basically starting at the same state as the *end* state in the previous test.